redis.go

```diff
@@ -118,10 +143,14 @@ local function isPaused(pauseKey)
   return redis.call('get', pauseKey)
 end
 
-local function canRun(lockKey, maxConcurrency)
+local function canRun(lockKey, lockInfoKey, maxConcurrency)
+  if (not maxConcurrency or maxConcurrency == 0) then
```

Comment:
This was moved to the top of the function to remove an unnecessary call to redis.